File: 303.range-sum-query-immutable.cpp
// CreateTime: 2021-01-19 00:19:27
class NumArray {
public:
    vector<int> p;

    NumArray(vector<int> nums) {
        p.resize(nums.size()+1);

        for (int i = 0; i < nums.size(); i++) {
            p[i+1] = p[i] + nums[i];
        }
    }

    int sumRange(int i, int j) {
        return p[j+1] - p[i];
    }
};

File: 303.range-sum-query-immutable.py
class NumArray(object):
    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.aux = []
        tot = 0

        for n in nums:
            tot += n
            self.aux.append(tot)


    def sumRange(self, i, j):
        """
        :type i: int
        :type j: int
        :rtype: int
        """
        if i == 0:
            return self.aux[j]

        return self.aux[j] - self.aux[i-1]
